### PR TITLE
Weld 3.0.0.Alpha16

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: news/2016-04-29-weld-300Alpha16.asciidoc [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+However, the main goal of this release is to align with CDI API 2.0.Alpha4, and in particular to implement the current version of "metadata builders" API (link:https://issues.jboss.org/browse/CDI-558[CDI-558, window="_blank"]).
+So that everyone can play with the API, discover possibilities and find potential issues.
+Note that this release also introduces a non-standard experimental feature: link:news/2016/02/08/weld-se-synth-lifecycle-events/[Weld SE synthetic container lifecycle event observers].
+So it should be even easier to get started - no extension class is needed in Weld SE.
+
+== Metadata configurators - basic ideas
+
+Note that we don't use the term "builder" because there is no `build()` method in the API.
+Also note that the API is not intended to cover every possible case.
+Instead, we would like help with common tasks.
```
Comment (manovotn) [inline news/2016-04-29-weld-300Alpha16.asciidoc]: `would like help` -> `would like to help`

File: news/2016-04-29-weld-300Alpha16.asciidoc [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+And if needed, an extension developer can always use the original replacements methods for more complicated stuff.
```
Comment (manovotn) [inline news/2016-04-29-weld-300Alpha16.asciidoc]: `replacements methods` -> `replacement methods`
`if needed` -> `if need be` / `if necessary`

File: news/2016-04-29-weld-300Alpha16.asciidoc [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+IMPORTANT POINTS:
+
+* an extension developer receives a configurator instance from a container lifecycle event
+* a configurator instance is always automatically processed at the end of the observer invocation
```
Comment (manovotn) [inline news/2016-04-29-weld-300Alpha16.asciidoc]: Maybe stress out that you can request several configurators within one observer invocation and that all will be automatically 'built' (case of BeanConfigurator)?

File: news/2016-04-29-weld-300Alpha16.asciidoc [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+** e.g. `ProcessAnnotatedType.configureAnnotatedType()` returns a configurator initialized with `ProcessAnnotatedType.getAnnotatedType()`
+** the result of the configurator will automatically replace the original component (e.g. `AnnotatedType` in case of `ProcessAnnotatedType`)
+** replacement methods (e.g. `ProcessAnnotatedType.setAnnotatedType()`) should not be used together with configurators (link:https://issues.jboss.org/browse/CDI-596[CDI-596, window="_blank"])
+* for **application lifecycle events** (e.g. `AfterBeanDiscovery`), the configurator is always uninitialized/empty, but we should probably define some default values wherever it makes sense (e.g. `Reception` for observer methods)
+** the result of the configurator will be automatically added (e.g. `Bean` in case of `AfterBeanDiscovery.addBean()`)
+
+We have prepared some simple examples - see below.
+More complicated examples can be found in link:https://github.com/cdi-spec/cdi-tck/tree/master/impl/src/main/java/org/jboss/cdi/tck/tests/extensions/configurators[TCK test cases, window="_blank"].
```
Comment (manovotn) [inline news/2016-04-29-weld-300Alpha16.asciidoc]: `More complicated examples` -> `More advanced examples`
Just not to disencourage ppl from those examples ;)

**Comment by mkouba** [comment]
Merged.

**Pull request closed without merging**